app/src/main/kotlin/com/bitwarden/authenticator/data/platform/util/JsonExtensions.kt

```diff
@@ -0,0 +1,19 @@
+package com.bitwarden.authenticator.data.platform.util
+
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.Json
+
+/**
+ * Attempts to decode the given JSON [string] into the given type [T]. If there is an error in
+ * processing the JSON or deserializing it to an instance of [T], `null` will be returned.
+ */
+inline fun <reified T> Json.decodeFromStringOrNull(
+    string: String,
+): T? =
+    try {
+        decodeFromString(string = string)
+    } catch (e: SerializationException) {
+        null
+    } catch (e: IllegalArgumentException) {
+        null
+    }
```

Comment:
🤔 Maybe it's being done elsewhere but as an improvement, could we have a logger here in debug mode when an exception is thrown? So we better understand what went wrong. Additionally, it'd be good to have the possibility to log into crashlytics by a parameter or something like that so we know there was an issue with decoding the string into the type.
This could be in a tech-debt task, not necessarily as part of this PR.

Reply:
There is currently no logging, but I do agree it is needed. We can track that as a future improvement. 👍